.github/workflows/pr-build.yml

```diff
@@ -9,16 +9,21 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     steps:
        # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
       - name: Check out code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           ref: ${{github.event.pull_request.head.ref}}
           repository: ${{github.event.pull_request.head.repo.full_name}}
           fetch-depth: 0
 
+      - uses: actions/setup-java@v4
```

Comment:
This repo has some incompatibilities with later versions of java that need to be fixed. Default java version moves to 17 as of ubuntu 24.04